<?xml version="1.0"?>

<launch>
	
	<arg name="robot_name"/>

	<group ns="$(arg robot_name)">

		<!-- localization -->
		<node name="fake_localization" pkg="fake_localization" type="fake_localization">
			<remap from="base_pose_ground_truth" to="odometry/gazebo" />
			<param name="odom_frame_id" type="string" value="$(arg robot_name)/odom"/>
			<param name="base_frame_id" type="string" value="$(arg robot_name)/base_link"/>
		</node>

		<!-- move_base -->
		<node name="move_base" pkg="move_base" type="move_base" output="screen">
			<remap from="odom" to="odometry/base_raw" />
			<remap from="cmd_vel" to="nav_vel" />

			<rosparam file="$(find assessment_package)/config/costmap_$(arg robot_name)_params.yaml" command="load" ns="global_costmap"/>
			<rosparam file="$(find assessment_package)/config/costmap_$(arg robot_name)_params.yaml" command="load" ns="local_costmap"/>

			<rosparam file="$(find assessment_package)/config/global_costmap_params.yaml" command="load"/>
			<rosparam file="$(find assessment_package)/config/local_costmap_params.yaml" command="load"/>

			<rosparam file="$(find assessment_package)/config/planners.yaml" command="load"/>
		</node>

	</group>

</launch>
